A4A: Removing the A4A environments from isJetpackCloud, adding the A4A environment check anywhere that it is needed

coder-karen opened the pull request:
Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/242

## Proposed Changes

* This PR removes the A4A environments from `client/lib/jetpack/is-jetpack-cloud.ts`. It also adds checks for the A4A environment with the newly created `isA8CForAgencies`, anywhere `isJetpackCloud` is used that appears relevant.
* This also includes a couple of small style tweaks as button styles in particular have been affected (largely due to the removal of the `is-jetpack-site` class).

What may need looking at later:
* We may need to create tests here that cover A4A aswell: `components/notes-formatted-block/test/blocks.js`. 
* We do not have an A4A platform yet to fetch eg. sites per user from:`components/site-selector/utils.ts`.
* `my-sites/controller.js` may need updating at some point when we have a plugins path, non JP cloud specific site selection considerations, and primary site needs.

There are references to `isJetpackCloud` that I've left adding checks for the a4a environment alongside. These predominantly include:
- We don't seem to use `VisibleDaysLimitUpsell`
- Jetpack social paths
- `JetpackComMasterbar` as we're not using it
- `Jetpackconnect`
- GA cross domain linking: `/lib/analytics/ga.js`
- Hotjar site settings: `/lib/analytics/hotjar.js`
- Other tracking: `/analytics/tracker-buckets.ts`, `analytics/ad-tracking/load-tracking-scripts.js`, `analytics/ad-tracking/setup.js`, `analytics/utils/get-ga-gtag.ts`
- `settingsBasePath` - we don't have one yet for A4A
- `PlanBillingPeriod` and `PaymentMethodBackupToggle`
- `/my-sites/backup/controller.js` - we have our own a4a version of this file
- Some checks in `/my-sites/backup/main.jsx` left as the current style relies on them
- Checkout thank you page URL redirect: `/my-sites/checkout/get-thank-you-page-url/index.ts`
- `/earn` path: `/my-sites/earn/controller.ts` and earn related files / components.
- `JetpackSearchContent` and anything related to Jetpack Search
- `/marketing/connections` related
- const `isJetpackSiteOrJetpackCloud` (sitePicker related)
- `/my-sites/plans/jetpack-plans/` related (we have our own license checkout process)
- `/my-sites/purchases` related
- `const notFoundIfNotEnabled` - this has not been modified for my-sites/backup and scan, as there is a commented note in this A4A file about it not working in A4A: `/a8c-for-agencies/sections/sites/features/jetpack/backup/routes.ts`
- `/my-sites/sidebar`
- `/my-sites/sites` (upgrade related)
- `/my-sites/subscribers`

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

Some of the changes have been difficult to test, but the below is testable:

* Test the preview pane Activity feed, eg. plugins and themes (and author) links will now be removed.
* Some classnames have been changed (eg. removing `wordpressdotcom` as an additional class in some cases) - so checking general styles aren't impacted would be good. Specifically: activity preview pane.
* In the preview pane, for any site under Activity, Backup, Scan that shows a time mismatch warning, the 'update' link should continue taking you to `wp-admin`, not `/settings/general/`
* Other things to test generally include Scan and Backup tabs in the preview pane (upsells, links, styles)
* Generally test functionality on the A4A dashboard to make sure nothing is broken. 
* Ideally also check Jetpack Cloud and WordPress.com environments (in particular with relation to Scan, Backup, Activity , Add new site buttons)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

coder-karen committed:
Adding checks for isA8CForAgencies anywhere there are checks for isJetpackCloud that appears relevant to A4A

coder-karen committed:
Fixing button styles related to changing the A4A environment additions from isJetpackCloud

gogdzl commented on client/components/site-selector/add-site.tsx:
```diff
@@ -11,16 +12,22 @@ const SiteSelectorAddSite: FunctionComponent = () => {
-			: 'calypso_add_new_wordpress_click';
+		let event = 'calypso_add_new_jetpack_click';
+		if ( ! isJetpackCloud() ) {
+			event = isA8CForAgencies() ? 'calypso_add_new_A4A_click' : 'calypso_add_new_wordpress_click';
```
If I recall correctly, we are using only lowercase characters for our tracks events, so I think this event should be called 'calypso_add_new_a4a_click'.

coder-karen replied:
Good point! Changed in f917e05c212a9adff5289dbfe60bc1bb17c03cf7

gogdzl commented on client/my-sites/activity/activity-log-v2/index.tsx:
```diff
@@ -34,7 +34,6 @@ import './style.scss';
 const ActivityLogV2: FunctionComponent = () => {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const isA4AEnabled = isEnabled( 'a8c-for-agencies' );
```
Nice chop =)

gogdzl reviewed (changes requested):
Everything seems to be working perfectly!

Just a small correction in our tracks event name, and below some nitpicks about styles (these are not blockers)

I noticed that we are losing these two styles for Backup: https://github.com/Automattic/wp-calypso/blob/9cc8a6e8c19b93b3d0713f249ad108421ef44218/client/components/jetpack/card/jetpack-product-card/display-price/style.scss#L14

We are also losing a margin style for the Activity log, but this one I think this PR's version looks better: https://github.com/Automattic/wp-calypso/blob/9cc8a6e8c19b93b3d0713f249ad108421ef44218/client/my-sites/activity/activity-log-v2/style.scss#L3

Also `split-button__main` looks different from staging, font-weight is set to 500, in staging font-weight is set to 400. I can't tell if this is intentional or not.

coder-karen committed:
Making sure tracks event name is lowercase

coder-karen committed:
Removing extra font-weight, currently it doesn't seem that font styles are being overwritten so this doesn't seem needed

coder-karen committed:
Adding font-weight back to 500 for buttons on overview page - but not in header

coder-karen committed:
Modifying overview contact support button styles to match those in staging

coder-karen commented:
Thanks for the review @gogdzl !

> I noticed that we are losing these two styles for Backup:
> Line 14 in [9cc8a6e](https://github.com/Automattic/wp-calypso/commit/9cc8a6e8c19b93b3d0713f249ad108421ef44218)

I don't see where we use Jetpack product cards though, or where the style has changed. Do you see an example of where this has changed on the Backup card - which kind of scenario in the Backup preview pane?

> We are also losing a margin style for the Activity log, but this one I think this PR's version looks better:
> Line 3 in [9cc8a6e](https://github.com/Automattic/wp-calypso/commit/9cc8a6e8c19b93b3d0713f249ad108421ef44218)

Ah good catch! Yes that was unintentional but the margin is more closely aligned with other tabs now so seems good like you say!

> Also split-button__main looks different from staging, font-weight is set to 500, in staging font-weight is set to 400. I can't tell if this is intentional or not.

Yes, button styles changed, it looks like I added the font weight too globally (to all buttons on the a8c theme). That's hopefully fixed again here by removing it: 92bc6b29b2acbf079243ea38ff5479e49a97548a. I've added font-weight of 500 specifically for the overview page buttons (not the header buttons), plus modifications to the overview contact support button now too.

coder-karen committed:
Adding a few additional button style changes throughout A4A to match staging

coder-karen committed:
Merge branch 'trunk' into update/a4a/add-a4a-check-alongside-jetpack-cloud-check

gogdzl commented:
> I don't see where we use Jetpack product cards though, or where the style has changed. Do you see an example of where this has changed on the Backup card - which kind of scenario in the Backup preview pane?

For some reason, now it looks better in this PR, Leave as it is.

Only for information, this is one of the pages these styles are applied:

**This PR**
![image](https://github.com/Automattic/wp-calypso/assets/37049295/ec130096-5d73-413c-82fb-803f9f3198d6)


**Staging**
![image](https://github.com/Automattic/wp-calypso/assets/37049295/e4e5565d-0507-40cf-a2ba-32a793e36c05)

gogdzl reviewed (approved):
Great work! 

LGTM!
